src/helper/waylandsocketwatcher.cpp

```diff
@@ -28,18 +28,20 @@ namespace SDDM {
 WaylandSocketWatcher::WaylandSocketWatcher(QObject *parent )
     : QObject(parent)
     , m_runtimeDir(QDir(QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation)))
-    , m_socketPath(m_runtimeDir.absoluteFilePath(QLatin1String("wayland-0")))
+    , m_socketName(QLatin1String("wayland-?"))
```

Comment:
I'd leave m_socketName as empty. It's a better way to signal we don't know the name. Don't want to return "wayland-?" on ::socketPath.

Reply:
It's there pretty much for logging purposes only - all these qDebugs printing what file the watcher is expecting.

There's a big shortcut I took with not separating m_socketName and m_socketPattern and not making the pattern configurable. I'd love to address that, but I can't figure out if there are ways to pass the value from the config to the helper other than setting the env variable in Greeter.cpp.
I'll do the first part and separate the variables.

(Wayland socket doesn't have to match `wayland-\d+` and some debug tools are actually using different value patterns.)